core/server/common/src/main/java/alluxio/master/journal/raft/JournalStateMachine.java

```diff
@@ -289,36 +305,108 @@ public CompletableFuture<Message> query(Message request) {
 
   @Override
   public void close() {
+    Exception e = new RuntimeException("Stop here and check");
+    e.printStackTrace();
+    LOG.error("JournalStateMachine is closing: ", e);
     mClosed = true;
   }
 
   @Override
   public CompletableFuture<Message> applyTransaction(TransactionContext trx) {
+    if (corrupted) {
+      LOG.error("The state machine is corrupted, stop here");
+    }
     try {
-      applyJournalEntryCommand(trx);
+      System.out.format("Master %s apply entry %s", mServer.getId(), trx.getLogEntry());
       RaftProtos.LogEntryProto entry = Objects.requireNonNull(trx.getLogEntry());
+      // TODO(Jiacheng): this is just a hack to simulate a crash condition
+      if (Configuration.getInt(PropertyKey.MASTER_EMBEDDED_JOURNAL_PORT) == 19202) {
+        // Fake a corruption around these indices
+        if (entry.getIndex() % 10 == 0 || (entry.getIndex() > 50 && entry.getIndex() < 70)) {
+          if (corrupted || recovered) {
+            LOG.info("Do not corrupt again after master recovery");
+          } else if (!corrupted) {
+            LOG.warn("Master {} is chosen to crash at index {}", 19202, entry.getIndex());
+//          corruptRatisServerInternally();
+            closeRatisServerInternally();
+            corrupted = true;
+            LOG.warn("The bomb is set to explode on the next log entry");
+            throw new AssertionError("What if I NPE here?");
+          }
+        }
+      }
+
```

Comment:
This is how I hack the internal ratis state bc it's really hard to introduce same errors to ratis crashes we see! It is just really hard bc ratis itself handles many errors. I need to just throw an error that cannot be caught! Also it has to crash ratis in a way that we want. There are many possible crashes:
1. An error that makes the ratis server exit normally (by triggering catch statements)
2. An error that makes the ratis server exit silently (by hitting a real ratis critical bug)